sklearn/metrics/tests/test_classification.py

```diff
@@ -394,6 +397,79 @@ def test_confusion_matrix_multiclass_subset_labels():
                             [24, 3]])
 
 
+def test_multilabel_confusion_matrix():
+    """Test multilabel confusion matrix"""
+
+    # Non Regression test
+    y_true = [[0, 1, 1, 1, 1],
+              [1, 0, 0, 0, 0],
+              [0, 0, 1, 1, 1],
+              [1, 1, 1, 0, 0],
+              [0, 0, 0, 0, 1],
+              [1, 0, 1, 0, 0]]
+
+    y_pred = [[1, 1, 0, 0, 1],
+              [1, 0, 0, 1, 0],
+              [0, 0, 0, 1, 1],
+              [1, 1, 0, 0, 1],
+              [0, 0, 0, 1, 1],
+              [1, 1, 1, 0, 0]]
+
+    # Check against pre computed multilabel confusion matrix
+    #                   TP FP TN FN
+    result = np.array([[3, 1, 0, 2],   # <-- Label 0
+                       [2, 1, 0, 3],
+                       [1, 0, 3, 2],
+                       [1, 2, 1, 2],
+                       [3, 1, 0, 2]])  # <-- Label 4
+
+    assert_array_equal(result, multilabel_confusion_matrix(y_true, y_pred))
+
+    # Test sparse input
+    y_true = sparse.csr_matrix(y_true)
+    y_pred = sparse.coo_matrix(y_pred)
+
+    assert_array_equal(result, multilabel_confusion_matrix(y_true, y_pred))
```

Comment:
To make the tests easier to follow, I think we should split the remainder of this function (after this line) to another function.

The first function could be named `test_multilabel_confusion_matrix_toy_data`, then another just for input validation (`assert_raises`) named `test_multilabel_confusion_matrix_errors` and then a function `test_multilabel_confusion_matrix_shape` for matrix slicing and shape checks.